ledger/persistedaccts_list.go

```diff
@@ -0,0 +1,127 @@
+package ledger
+
+// persistedAccountDataList represents a doubly linked list.
+// must initiate with newPersistedAccountList.
+type persistedAccountDataList struct {
+	root     persistedAccountDataListNode // sentinel list element, only &root, root.prev, and root.next are used
+	len      int                          // current list length excluding (this) sentinel element
+	freeList *persistedAccountDataList    // preallocated nodes location
```

Comment:
I think that having a simple `* persistedAccountDataListNode` is going to be more performant here.

Reply:
i've changed it, though i haven't notice a change in performance.